scripts/check_site_lib.lua

```diff
@@ -74,8 +74,10 @@ function need_array(varname, subcheck, required)
 
    assert_type(var, 'table', "site.conf error: expected `" .. varname .. "' to be an array")
 
-   for _, e in ipairs(var) do
-      subcheck(e)
+   if subcheck then
+      for _, e in ipairs(var) do
+         subcheck(e)
+      end
```

Comment:
@NeoRaider: I currently left this check, because I use need_array() to get the array of supported_rates to compare against mcast_rate and basic_rate. Maybe there is another solution to solve this.